ui-frontend/packages/catalog-ui-search/src/main/webapp/react-component/utils/export/export.tsx

```diff
@@ -108,3 +132,202 @@ export const exportResultSet = async (
     },
   })
 }
+
+export const getWarning = (exportCountInfo: ExportCountInfo): string => {
+  const exportCount = getExportCount(exportCountInfo)
+  const result = exportCountInfo.selectionInterface
+    .getCurrentQuery()
+    .get('result')
+  const totalHits = getHits(Object.values(result.get('lazyResults').status))
+  const limitWarning = `You cannot export more than the administrator configured limit of ${
+    (properties as any).exportResultLimit
+  }.`
+  let warningMessage = ''
+  if (exportCount > (properties as any).exportResultLimit) {
+    if (exportCountInfo.exportSize === 'custom') {
+      return limitWarning
+    }
+    warningMessage =
+      limitWarning +
+      `  Only ${(properties as any).exportResultLimit} ${
+        (properties as any).exportResultLimit === 1 ? `result` : `results`
+      } will be exported.`
+  }
+  if (exportCountInfo.exportSize === 'custom') {
+    if (exportCount > totalHits) {
+      warningMessage = `You are trying to export ${exportCount} results but there ${
+        totalHits === 1 ? `is` : `are`
+      } only ${totalHits}. \nOnly ${totalHits} ${
+        totalHits === 1 ? `result` : `results`
+      } will be exported.`
+    }
+  }
+  if (
+    totalHits > 100 &&
+    exportCount > 100 &&
+    (properties as any).exportResultLimit > 100
+  ) {
+    warningMessage += `  This may take a long time.`
+  }
+  return warningMessage
+}
+
+export const generateOnDownloadClick = ({
+  addSnack,
+}: {
+  addSnack: AddSnack
+}) => {
+  return async (downloadInfo: DownloadInfo) => {
+    const exportFormat = encodeURIComponent(downloadInfo.exportFormat)
+    try {
+      const body = getDownloadBody(downloadInfo)
+      const response = await exportResultSet(exportFormat, body)
+      if (response.status === 200) {
+        const data = await response.blob()
+        const contentType = response.headers.get('content-type')
+        const filename = contentDisposition.parse(
+          response.headers.get('content-disposition')
+        ).parameters.filename
+        saveFile(filename, 'data:' + contentType, data)
+      } else {
+        addSnack('Error: Could not export results.', {
+          alertProps: { severity: 'error' },
+        })
+      }
+    } catch (error) {
+      console.error(error)
+    }
+  }
+}
+
+function getStartIndex(src: string, selectionInterface: any) {
+  const srcIndexMap = selectionInterface.getCurrentQuery()
+    .nextIndexForSourceGroup
+  if (src === Sources.localCatalog) {
+    return srcIndexMap['local']
+  }
+  return srcIndexMap[src]
+}
+
+function getSrcs(selectionInterface: any) {
+  return selectionInterface.getCurrentQuery().getSelectedSources()
+}
+
+function getSrcCount(src: any, selectionInterface: any) {
+  const result = selectionInterface.getCurrentQuery().get('result')
+  return Object.values(
+    result.get('lazyResults').status as {
+      [key: string]: any
+    }
+  ).find((status: any) => status.id === src).count
```

Comment:
I dont think this `find()` will ever return `undefined` since `src` is from `srcs = selectionInterface.getCurrentQuery().getSelectedSources()`? Or can it still?